Build the docs with the latest version of Sphinx

asmeurer opened the pull request:
Fixes #12841 and #12958.

asmeurer committed:
Remove unused import in numpydoc.py

It causes an error with the development version of Sphinx.

asmeurer committed:
Suppress citation warnings in Sphinx

asmeurer committed:
Suppress footnote warnings

For some reason some of the references in the solveset docs are being seen as
unused footnotes.

asmeurer committed:
Remove make html-errors from docs build

Now make html always errors on warnings.

asmeurer committed:
Build the docs with the latest version of Sphinx and docutils

This should fix #12958 (search is broken in the docs).

moorepants commented:
Sphinx now ships with the napoleon extension which can replace the numpydoc extension. Might be worth switching now.

asmeurer committed:
Install latexmk in Travis for building the docs

asmeurer commented:
I'm not up for doing it. Let's open a new issue about it.

asmeurer committed:
Fix sed command to work on Mac

asmeurer committed:
Fix latexmk to use xetex in docs build

asmeurer committed:
Fix footnote in Heaviside docstring

Sphinx treats all footnotes as references (if otherwise, I don't know how to
make it work). The * was breaking the latex build.

asmeurer committed:
Fix sed command in docs Makefile

asmeurer committed:
Use dist: trusty in Travis

This will hopefully fix the LaTeX build issues. Also, it will eventually be
the default anyway
(https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming).

asmeurer commented:
Yes, it worked!
